core/commands/MonitorCommand.php

```diff
@@ -0,0 +1,41 @@
+<?php
+# TODO: Do autoload for commands.
+require_once( dirname( __FILE__ ) . '/Command.php' );
+
+class MonitorCommand extends Command {
+	function __construct( array $p_data, Context $p_context ) {
+		parent::__construct( $p_data, $p_context );
+	}
+
+	function validate() {
+		# Validate issue id
+		if( !isset( $this->data['issue_id'] ) ) {
+			throw new CommandException( HTTP_STATUS_BAD_REQUEST, 'issue_id missing', ERROR_GPC_VAR_NOT_FOUND );
```

Comment:
It might be worth extracting out a InvalidDataCommandException or similar that returns the 400 Bad Request status, since it's going to be a very often encountered situation.

Reply:
Makes sense.  Will revisit the exception hierarchy or having helper methods to create such exception instances.  I have already done that for the REST/SOAP APIs.